core/src/main/java/hudson/model/StreamBuildListener.java

```diff
@@ -80,5 +93,41 @@ public void finished(Result result) {
         getLogger().println("Finished: "+result);
     }
 
+    @Override
+    public void close() throws IOException {
+        try {
+            super.close();
+        } catch (IOException e) {
+            if (additionalClosables != null) {
+                for (Closeable c: additionalClosables) {
+                    try {
+                        c.close();
+                    } catch (IOException e1) {
+                        e.addSuppressed(e1);
```

Comment:
If we are to do something this fancy, I think it should be a more general-purpose utility, as proposed here: https://github.com/jenkinsci/remoting/pull/104#r75861893